src/arviz_plots/plots/utils.py

```diff
@@ -168,14 +169,68 @@ def set_grid_layout(pc_kwargs, plot_bknd, ds, num_rows=None, num_cols=None):
     return pc_kwargs
 
 
+def _references_to_dataset(references, ds, sample_dims):
+    """Generate an :class:`~xarray.Dataset` compatible with `ds` from `references`.
+
+    Parameters
+    ----------
+    references : scalar, array-like, dict, DataArray
+        References to cast into a compatible dataset.
+    ds : Dataset
+    sample_dims : iterable of hashable, optional
+
+    Returns
+    -------
+    Dataset
+        A Dataset with a subset of the variables, dimensions and coordinate names in `ds`,
+        with only an extra dimension "ref_line_dim" added when multiple references are requested
+        for one or some of the variables.
+    """
+    if isinstance(references, xr.Dataset):
+        return references
+    if isinstance(references, xr.DataArray):
+        name = references.name
+        if name is not None:
+            if name not in ds.data_vars:
+                warnings.warn(f"{name} not available in {ds.data_vars}", UserWarning)
+            return references.to_dataset()
+        references = references.values
+    if np.isscalar(references):
+        aux_ds = (
+            ds if not sample_dims else ds.isel({dim: 0 for dim in sample_dims if dim in ds.dims})
+        )
+        return xr.full_like(aux_ds, references, dtype=np.array(references).dtype)
+    if isinstance(references, list | tuple | np.ndarray):
+        references = {var_name: references for var_name in ds.data_vars}
+    if isinstance(references, dict):
+        ref_dict = {}
+        for var_name, da in ds.items():
+            if var_name not in references:
+                continue
+            ref_values = references[var_name]
+            sizes = {dim: length for dim, length in da.sizes.items() if dim not in sample_dims}
+            ref_dict[var_name] = xr.DataArray(
+                np.full(list(sizes.values()) + [np.size(ref_values)], ref_values),
+                dims=list(sizes) + ["ref_line_dim"],
+                coords={"ref_line_dim": np.arange(np.size(ref_values))}
+                | {
+                    coord_name: coord_da
+                    for coord_name, coord_da in da.coords.items()
+                    if coord_da.dims[0] not in sample_dims
+                },
+            )
+        return xr.Dataset(ref_dict)
+    raise TypeError("Unrecognized input type for `references`")
+
+
 def add_reference_lines(
     plot_collection,
     references,
     orientation="vertical",
     aes_map=None,
     plot_kwargs=None,
     backend=None,
```

Comment:
Given `plot_collection` is a required argument, backend should not be an argument at all, the backend should be retrieved from the plot_collection to make sure both are aligned, otherwise plotting won't work.